default.nix

```diff
@@ -26,46 +30,56 @@ let
     noTest = pkg: haskell.lib.dontCheck pkg;
 
     hspkgs = haskell.packages.${bootghc};
+    ghc    = haskell.compiler.${bootghc};
 
     ourtexlive =
       nixpkgs.texlive.combine {
         inherit (nixpkgs.texlive) scheme-small collection-xetex fncychap titlesec tabulary varwidth framed capt-of wrapfig needspace dejavu-otf helvetic; };
     fonts = nixpkgs.makeFontsConf { fontDirectories = [ nixpkgs.dejavu_fonts ]; };
     docsPackages = if withDocs then [ python3Packages.sphinx ourtexlive ] else [];
 
-    deps =
-      [ autoconf automake m4
+    depsSystem = with stdenv.lib; (
+      [ arcanist
+        autoconf automake m4
         gmp.dev gmp.out glibcLocales
         ncurses.dev ncurses.out
         perl git file which python3
-        (hspkgs.ghcWithPackages (ps: [ ps.alex ps.happy ]))
         xlibs.lndir  # for source distribution generation
-        cabal-install
         zlib.out
         zlib.dev
       ]
       ++ docsPackages
-      ++ stdenv.lib.optional withLlvm llvm_6
-      ++ stdenv.lib.optional withNuma numactl
-      ++ stdenv.lib.optional withDwarf elfutils
-      ++ stdenv.lib.optional (! stdenv.isDarwin) pxz ;
+      ++ optional  withLlvm            llvm_6
```

Comment:
While you're cooking up a patch... could you switch that to `llvm_7`, the version needed to build HEAD? Thanks!